yaml/src/main/python/main.py

```diff
@@ -58,6 +45,28 @@ def _get_pipeline_yaml():
         pipeline_yaml = fin.read().decode()
     return pipeline_yaml
 
+def _parse_yaml_pipeline(yaml_pipeline):
+    """Parses a YAML pipeline string to extract Jinja variable names.
+
+    This function uses a regular expression to find all occurrences of Jinja-style
+    placeholders (e.g., `{{ my_var | default('foo') }}`) and extracts the core
+    variable name, ignoring filters and surrounding whitespace.
+
+    Args:
+        yaml_pipeline (str): The YAML pipeline content as a string.
+
+    Returns:
+        set: A set of unique Jinja variable names found in the pipeline.
+    """
+    pattern = r'\{\{(.*?)\}\}'
```

Comment:
Rather than using a regex, can we just:

1) Parse the yaml with a library? We should be able to use `yaml.load` like we do in Beam using the `pyyaml` library
2) Iterate over the parameters section

That seems safer/less error prone (I also assumed this was the main value of having a parameters block).

Reply:
Went with jinja2 library that handles finding the variables.  Thanks

Reply:
Thanks, that's a good idea